src/sqlfluff/dialects/dialect_exasol.py

```diff
@@ -3770,3 +3766,17 @@ class FileSegment(BaseFileSegment):
             allow_trailing=True,
         ),
     )
+
+
+@exasol_dialect.segment()
+class EmitsGrammar(BaseFileSegment):
```

Comment:
```suggestion
class EmitsGrammar(BaseSegment):
```
`BaseFileSegment` should only be used by `FileSegment`

Reply:
Oops - good spot. Fixed